packages/@ember/-internals/views/lib/views/states.ts

```diff
@@ -2,6 +2,15 @@ import preRender from './states/pre_render';
 import hasElement from './states/has_element';
 import inDOM from './states/in_dom';
 import destroying from './states/destroying';
+import Component from '@ember/component';
+
+export interface ViewState {
+  enter?(view: Component): void;
+  appendChild(): void;
+  handleEvent(view: Component, eventName: string, event: Event): boolean;
+  rerender(view: Component): void;
+  destroy(view: Component): void;
```

Comment:
I don't love that the `view` type here is `Component`. It feels like it should be more generic. Alternatively, we should stop using the generic `View` and `CoreView` types.

Reply:
I don't even actually know what the relationship between `View`, `CoreView`, and `Component` actually is! 😬